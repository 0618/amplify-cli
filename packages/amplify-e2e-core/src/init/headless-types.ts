/**
 * Shape of awscloudformation object  payload for init/pull
 */
export type AwsProviderConfig = AwsProviderProfileConfig | AwsProviderGeneralConfig;

/**
 * Shape of awscloudformation object within `--profile` payload for init/pull
 */
export type AwsProviderProfileConfig = {
  configLevel: string;
  useProfile: boolean;
  profileName: string;
};

/**
 * Shape of awscloudformation object when general config is used for init/pull
 */

export type AwsProviderGeneralConfig = {
  configLevel: string;
};

/**
 * amplify profile config type
 */
export type AmplifyProfileConfig = 'accessKeys' | 'profile' | 'general';

/**
 * Shape of `--categories` payload for init/pull
 */
export type CategoriesConfig = Record<string, unknown>;
