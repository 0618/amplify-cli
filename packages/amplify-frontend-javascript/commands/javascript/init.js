module.exports = {
    name: 'init',
    run: async (context) => {
        console.log('int js project///')
    }
}
  