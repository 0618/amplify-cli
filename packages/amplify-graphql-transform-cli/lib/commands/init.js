"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const clime_1 = require("clime");
const fs = require("fs");
const log_1 = require("../log");
const TemplateSchema = `type Post @model {
    id: ID!
    title: String! @search
    tags: [String] @search
    createdAt: String
    updatedAt: String
}
`;
const DefaultConfig = (name) => `
{
    "name": "${name}"
}
`;
let default_1 = class default_1 extends clime_1.Command {
    execute() {
        /**
         * Initialize an application.
         */
        log_1.default.debug('Creating schema.graphql');
        fs.writeFileSync('schema.graphql', TemplateSchema);
        log_1.default.info(`
            Edit the schema.graphql file and then run 'appsync create schema.graphql <application-name>' to build and deploy your API.
        `);
    }
};
__decorate([
    clime_1.metadata,
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], default_1.prototype, "execute", null);
default_1 = __decorate([
    clime_1.command({
        description: 'Deploy an AppSync API from your schema.graphql file',
    })
], default_1);
exports.default = default_1;
//# sourceMappingURL=init.js.map