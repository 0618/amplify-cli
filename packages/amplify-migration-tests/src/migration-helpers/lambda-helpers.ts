import {
  ExecutionContext,
  FunctionRuntimes,
  getRuntimeDisplayName,
  getTemplateChoices,
  moveUp,
  runtimeChoices,
  singleSelect,
} from '@aws-amplify/amplify-e2e-core';

export const selectRuntimePreV12 = (chain: ExecutionContext, runtime: FunctionRuntimes) => {
  const runtimeName = getRuntimeDisplayName(runtime);
  chain.wait('Choose the runtime that you want to use:');

  // reset cursor to top of list because node is default but it throws off offset calculations
  moveUp(chain, runtimeChoices.indexOf(getRuntimeDisplayName('nodejs')));

  singleSelect(chain, runtimeName, runtimeChoices);
};

export const selectTemplatePreV12 = (chain: ExecutionContext, functionTemplate: string, runtime: FunctionRuntimes) => {
  const templateChoices = getTemplateChoices(runtime);
  chain.wait('Choose the function template that you want to use');

  // reset cursor to top of list because Hello World is default but it throws off offset calculations
  moveUp(chain, templateChoices.indexOf('Hello World'));

  singleSelect(chain, functionTemplate, templateChoices);
};
