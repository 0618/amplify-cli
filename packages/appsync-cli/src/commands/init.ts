import { Command, command, param, metadata } from 'clime';
import File from '../types/File';
import GraphQLTransform from 'graphql-transform';
import SimpleTransform from 'simple-appsync-transform'
import * as fs from 'fs'
import log from '../log'

const TemplateSchema = `type Post @model {
    id: ID!
    title: String! @search
    tags: [String] @search
    createdAt: String
    updatedAt: String
}
`
const DefaultConfig = (name: string) => `
{
    "name": "${name}"
}
`

@command({
    description: 'Deploy an AppSync API from your schema.graphql file',
})
export default class extends Command {
    @metadata
    public execute() {
        /**
         * Initialize an application.
         */
        log.debug('Creating schema.graphql')
        fs.writeFileSync('schema.graphql', TemplateSchema)
        log.info(`
            Edit the schema.graphql file and then run 'appsync create schema.graphql <application-name>' to build and deploy your API.
        `)
    }
}
