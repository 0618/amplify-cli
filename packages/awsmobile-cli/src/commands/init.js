const analyzeProject = require('../lib/init-steps/s0-analyzeProject'); 
const configSettings = require('../lib/init-steps/s1-configSettings'); 
const initProviders = require('../lib/init-steps/s2-initProviders'); 
const onFailure = require('../lib/init-steps/s9-onFailure'); 
const onSuccess = require('../lib/init-steps/s9-onSuccess'); 

module.exports = {
    name: 'init',
    run: async (context) => {
        analyzeProject.run(context)
        .then(configSettings.run)
        .then(initProviders.run)
        .then(onSuccess.run)
        .catch(onFailure); 
        
    }
}
  