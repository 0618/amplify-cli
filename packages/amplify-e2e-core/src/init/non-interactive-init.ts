import execa from 'execa';
// eslint-disable-next-line import/no-cycle
import { getCLIPath, TEST_PROFILE_NAME } from '..';
import { CategoriesConfig, AwsProviderConfig } from './headless-types';

/**
 * Executes a non-interactive init to attach a local project to an existing cloud environment
 */
export const nonInteractiveInitAttach = async (
  projRoot: string,
  amplifyInitConfig: AmplifyInitConfig,
  categoriesConfig?: CategoriesConfig,
  awsProviderConfig = getAwsProviderConfig(),
): Promise<void> => {
  const args = [
    'init',
    '--yes',
    '--amplify',
    JSON.stringify(amplifyInitConfig),
    '--providers',
    JSON.stringify({
      awscloudformation: awsProviderConfig,
    }),
  ];
  if (categoriesConfig) {
    args.push('--categories', JSON.stringify(categoriesConfig));
  }
  await execa(getCLIPath(), args, { cwd: projRoot });
};


/**
 * Executes a non-interactive init to create a new Project
 */
export const headlessInit = async (
  projRoot: string,
  amplifyInitConfig: AmplifyInitConfig,
  frontendConfig: AmplifyFrontend,
  awsProviderConfig: AwsProviderConfig | AwsProviderGeneralConfig,
): Promise<void> => {
  const args = [
    'init',
    '--amplify',
    JSON.stringify(amplifyInitConfig),
    '--frontend',
    JSON.stringify(frontendConfig),
    '--providers',
    JSON.stringify({
      awscloudformation: awsProviderConfig,
    }),
    '--yes',
  ];
  await execa(getCLIPath(true), args, { cwd: projRoot });
};

/**
 * Executes a non-interactive init to migrate a local project to an existing cloud environment with forcePush flag
 */
export const nonInteractiveInitWithForcePushAttach = async (
  projRoot: string,
  amplifyInitConfig: AmplifyInitConfig,
  categoriesConfig?: CategoriesConfig,
  testingWithLatestCodebase = false,
  awsProviderConfig = getAwsProviderConfig(),
): Promise<void> => {
  const args = [
    'init',
    '--yes',
    '--amplify',
    JSON.stringify(amplifyInitConfig),
    '--providers',
    JSON.stringify({
      awscloudformation: awsProviderConfig,
    }),
    '--forcePush',
  ];
  if (categoriesConfig) {
    args.push('--categories', JSON.stringify(categoriesConfig));
  }
  await execa(getCLIPath(testingWithLatestCodebase), args, { cwd: projRoot });
};

/**
 * Returns an AmplifyConfig object with a default editor
 */
export const getAmplifyInitConfig = (projectName: string, envName: string): AmplifyInitConfig => ({
  projectName,
  envName,
  defaultEditor: 'code',
});

/**
 * Returns an frontend Config object of passed frontend type
 */

export const getAmplifyFrontend = () : AmplifyFrontend => ({
  frontend:    'javascript',
  framework:   'react',
  config:   getJavascriptConfig(),
});

const getJavascriptConfig = (): JavaScriptConfig => ({
    BuildCommand:   'npm run build',
    DistributionDir:   'testDist',
    SourceDir:  'src',
    StartCommand:  'npm run start'
});

/**
 * Returns a default AwsProviderConfig
 */
export const getAwsProviderConfig = (): AwsProviderConfig => ({
  configLevel: 'project',
  useProfile: true,
  profileName: TEST_PROFILE_NAME,
});

/**
 * Returns a general AwsProviderConfig
 */
export const getAwsProviderGeneralConfig = (): AwsProviderGeneralConfig => ({
  configLevel: 'general',
});

/**
 * Shape of `--amplify` payload for init/pull
 */
export type AmplifyInitConfig = {
  projectName: string;
  envName: string;
  defaultEditor: string;
  frontend?: string;
};

type JavaScriptConfig = {
  SourceDir :string;
  DistributionDir : string;
  BuildCommand : string;
  StartCommand : string;
}

type AmplifyFrontend = {
  frontend: string;
  framework: string;
  config: JavaScriptConfig;
}

export type AwsProviderGeneralConfig = {
  configLevel: string;
};
