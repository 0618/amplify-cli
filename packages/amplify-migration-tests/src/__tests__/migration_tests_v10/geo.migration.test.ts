import {
  addAuthWithDefault,
  addGeofenceCollectionWithDefault,
  addMapWithDefault,
  addPlaceIndexWithDefault,
  amplifyPull,
  amplifyPushWithoutCodegen,
  createNewProjectDir,
  deleteProject,
  deleteProjectDir,
  getAppId,
  updateAuthAddUserGroups,
} from '@aws-amplify/amplify-e2e-core';
import { versionCheck, allowedVersionsToMigrateFrom } from '../../migration-helpers';
import { initJSProjectWithProfileV10 } from '../../migration-helpers-v10/init';
import {
  assertNoParameterChangesBetweenProjects,
  collectCloudformationDiffBetweenProjects,
} from '../../migration-helpers/utils';

describe('geo category migration from v10 to latest', () => {
  const projectName = 'geoMigration';
  let projRoot: string;

  beforeAll(async () => {
    const migrateFromVersion = { v: 'unintialized' };
    const migrateToVersion = { v: 'unintialized' };
    await versionCheck(process.cwd(), false, migrateFromVersion);
    await versionCheck(process.cwd(), true, migrateToVersion);
    console.log(`Test migration from: ${migrateFromVersion.v} to ${migrateToVersion.v}`);
    expect(migrateFromVersion.v).not.toEqual(migrateToVersion.v);
    expect(allowedVersionsToMigrateFrom).toContain(migrateFromVersion.v);
  });

  beforeEach(async () => {
    projRoot = await createNewProjectDir(projectName);
    await initJSProjectWithProfileV10(projRoot, { name: 'geoMigration', disableAmplifyAppCreation: false });
    await addAuthWithDefault(projRoot);
    await addMapWithDefault(projRoot);
    await addPlaceIndexWithDefault(projRoot);
    const cognitoGroups = ['admin', 'admin1'];
    await updateAuthAddUserGroups(projRoot, cognitoGroups);
    await addGeofenceCollectionWithDefault(projRoot, cognitoGroups);
    await amplifyPushWithoutCodegen(projRoot);
  });

  afterEach(async () => {
    await deleteProject(projRoot, null, true);
    deleteProjectDir(projRoot);
  });

  it('...pull and push should not drift with new amplify version', async () => {
    const appId = getAppId(projRoot);
    expect(appId).toBeDefined();
    const projRoot2 = await createNewProjectDir('geoMigration2');
    try {
      await amplifyPull(projRoot2, { emptyDir: true, appId }, true);
      assertNoParameterChangesBetweenProjects(projRoot, projRoot2);
      expect(collectCloudformationDiffBetweenProjects(projRoot, projRoot2)).toMatchSnapshot();
      await amplifyPushWithoutCodegen(projRoot2, true);
      assertNoParameterChangesBetweenProjects(projRoot, projRoot2);
      expect(collectCloudformationDiffBetweenProjects(projRoot, projRoot2)).toMatchSnapshot();
    } finally {
      deleteProjectDir(projRoot2);
    }
  });
});
