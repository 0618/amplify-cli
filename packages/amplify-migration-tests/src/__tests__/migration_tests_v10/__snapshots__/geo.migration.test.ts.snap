// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`geo category migration from v10 to latest ...pull and push should not drift with new amplify version 1`] = `""`;

exports[`geo category migration from v10 to latest ...pull and push should not drift with new amplify version 2`] = `""`;
